### PR TITLE
bugfix: ZENKO-1522 fix CRR on copied MPU object

### PR DESCRIPTION
**Pull request by jonathan-gramain**
Change the logic to ignore the 'MPU' flag set in replicationInfo.content array (it may not always be set), instead rely on the object size and if a MPU or not based on its content-md5 metadata attribute containing a dash.

Note: the copy code lacks integrity checking now, this would need a
rework but it's out of scope of this fix.

### CODE REVIEW

**Comment by jonathan-gramain** [comment]
@bert-e reset

File: extensions/replication/tasks/MultipleBackendTask.js [hunk 1]
```diff
@@ -825,7 +799,26 @@ class MultipleBackendTask extends ReplicateObject {
                     return this._deleteObjectTagging(sourceEntry, destEntry,
                         log, next);
                 }
-                return this._getAndPutData(sourceEntry, destEntry, log, next);
+                // Do a multipart upload when either the size is above
+                // a threshold or the source object is itself a MPU.
```
Comment (bennettbuchanan) [inline extensions/replication/tasks/MultipleBackendTask.js:803]: What is the rationale for the 20MB threshold?

**Review by bennettbuchanan — APPROVED** [review]

File: extensions/replication/tasks/MultipleBackendTask.js (hunk 1 above)
Comment (jonathan-gramain) [reply extensions/replication/tasks/MultipleBackendTask.js:803]: It's a heuristic, if the object is "big enough" we want to MPU it to benefit from more parallelism (and later maybe better recovery on error by keeping the valid parts around), if the object is too small it's overkill to do a MPU (more requests and metadata overhead). So we need some boundary to decide between the two, 20MB makes a MPU of 4x5MB parts for example, which sounds like a reasonable size to start thinking about MPU.

Of course the size threshold can be debated, it just looks a reasonable tradeoff to me.

**Comment by jonathan-gramain** [comment]
There is a new test in E2E as well: https://github.com/scality/Zenko/compare/bugfix/ZENKO-1522-copyObjectMPUCRRTest

**Review by rahulreddy — APPROVED** [review]

**Review by rahulreddy — COMMENTED** [review]
Make sure you commit the package.lock changes as well.

**Review by ploki — APPROVED** [review]

**Comment by jonathan-gramain** [comment]
@bert-e reset

**Comment by jonathan-gramain** [comment]
@bert-e approve

**Comment by jonathan-gramain** [comment]
@bert-e reset

**Pull request merged**